modules/client/src/main/java/org/apache/ignite/internal/client/compute/ClientCompute.java

```diff
@@ -90,108 +92,128 @@ public ClientCompute(ReliableChannel ch, ClientTables tables) {
     }
 
     @Override
-    public <R> JobExecution<R> submit(JobTarget target, JobDescriptor descriptor, Object... args) {
+    public <T, R> JobExecution<R> submit(JobTarget target, JobDescriptor<T, R> descriptor, T args) {
```

Comment:
Here and in other places
```suggestion
    public <T, R> JobExecution<R> submit(JobTarget target, JobDescriptor<T, R> descriptor, T arg) {
```